### ISSUE TITLE
Implement --max-filesize curl option

### ISSUE THREAD

**Issue by jcamiel** [Orange-OpenSource/hurl#2353]
With --output user can download response to file system. Implementing --max-filesize can help preventing download of too big files.

**Comment by baerwang** [comment]
when will you review my code?

**Comment by fabricereix** [comment]
you need to sign your commits
![image](https://github.com/Orange-OpenSource/hurl/assets/682123/962cd470-8bba-4688-9add-22715efa6aa1)

**Comment by fabricereix** [comment]
if the max filesize is reached, Hurl should probably exit in error to be consistent with curl.
```
$ curl --max-filesize 10 https://hurl.dev
curl: (63) Maximum file size exceeded
$ echo $?
63
```

**Comment by jcamiel** [comment]
Fixed by #2468
